signer/src/testing/request_decider.rs

```diff
@@ -0,0 +1,541 @@
+//! Test utilities for the transaction signer
+
+use std::collections::BTreeSet;
+use std::time::Duration;
+
+use crate::context::Context;
+use crate::context::RequestDeciderEvent;
+use crate::context::SignerEvent;
+use crate::context::SignerSignal;
+use crate::error::Error;
+use crate::keys::PrivateKey;
+use crate::keys::PublicKey;
+use crate::message::Payload;
+use crate::network::in_memory2::SignerNetwork;
+use crate::network::in_memory2::SignerNetworkInstance;
+use crate::network::in_memory2::WanNetwork;
+use crate::network::MessageTransfer as _;
+use crate::request_decider::RequestDeciderEventLoop;
+use crate::storage;
+use crate::storage::model;
+use crate::storage::DbRead;
+use crate::storage::DbWrite;
+use crate::testing;
+use crate::testing::storage::model::TestData;
+
+use hashbrown::HashSet;
+use rand::SeedableRng as _;
+use tokio::sync::broadcast;
+use tokio::time::error::Elapsed;
+
+use super::context::*;
+
+/// A test harness for the signer event loop.
```

Comment:
nit: update comment

Reply:
Fixed in https://github.com/stacks-network/sbtc/pull/872/commits/abb9db7529ca6ce90001ca9433bf0ca72e675117